### PR TITLE
Change CI from using `cspell` to `codespell`

### PR DESCRIPTION
**Pull request by pplantinga**
The `cspell` tool comes from the `node.js` ecosystem and apparently fails to install on some systems due to requiring a fairly recent version of `GLIBC`, see this issue for more details:

https://github.com/jupyterlab/jupyterlab/issues/12675

This PR suggests we switch to `codespell` which is in python. The tool is less mature but still has > 1k stars and should be well-supported. In addition, it found a bunch more spelling errors.

### CODE REVIEW

File: recipes/Fisher-Callhome-Spanish/fisher_callhome_prepare.py [hunk 1]
```diff
@@ -710,7 +710,7 @@ def remove_labels(transcription: str):
     transcription = re.sub(r"<foreign langenglishonline", "", transcription)
     transcription = re.sub(r"<foreign langenglish", "", transcription)
     transcription = re.sub(r"</foreign", "", transcription)
-    transcription = re.sub(r"<[/]?foreing\s*\w*>", "", transcription)
+    transcription = re.sub(r"<[/]?foreign\s*\w*>", "", transcription)
```
Comment (Adel-Moumen) [inline recipes/Fisher-Callhome-Spanish/fisher_callhome_prepare.py:713]: We should make sure that this regex is still valid

**Review by Adel-Moumen — COMMENTED** [review]
ok, good! I left one comment. BTW, could you please fetch the latest speechbrain update? I suspect that you will need to run the linter on the new merged PR as well. I will check your PR locally once it's ready . Thanks again :)

File: recipes/Fisher-Callhome-Spanish/fisher_callhome_prepare.py (hunk 1 above)
Comment (pplantinga) [reply recipes/Fisher-Callhome-Spanish/fisher_callhome_prepare.py:713]: If you look at the context (before and after) all the lines are "foreign" so I'm pretty confident this one was supposed to be as well.

Comment (pplantinga) [reply recipes/Fisher-Callhome-Spanish/fisher_callhome_prepare.py:713]: Unless there's bad tags in the data that this was removing

**Review by Adel-Moumen — APPROVED** [review]
LGTM! Works on my lab cluster now.

**Pull request merged**